src/Categories/Category/Construction/KaroubiEnvelope.agda

```diff
@@ -0,0 +1,71 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category.Core
+
+-- Karoubi Envelopes. These are the free completions of categories under split idempotents
+module Categories.Category.Construction.KaroubiEnvelope {o ℓ e} (𝒞 : Category o ℓ e) where
+
+open import Level
+
+open import Categories.Morphism.Idempotent 𝒞
+open import Categories.Morphism.Reasoning 𝒞
+
+open Category 𝒞
+open HomReasoning
+open Equiv
+
+-- It's nicer to work with the bundled form of idempotents here.
+-- We could use Σ types, but that gets a bit messy.
+record BundledIdempotent : Set (o ⊔ ℓ ⊔ e) where
+  field
+    {obj} : Obj
+    isIdempotent : Idempotent obj
+
+  open Idempotent isIdempotent public
+
+record Idempotent⇒ (I J : BundledIdempotent) : Set (ℓ ⊔ e) where
```

Comment:
May as well move this too.  Though I think maybe in a submodule (with the above).  `.Bundled`? `.Structure`? There's an open issue where this naming is discussed.

Reply:
I've placed it in `Bundles` to keep in line with the stdlib naming conventions

Reply:
Ok, good.  We'll have to rename some others to follow suit.